src/screens/AddMedicineScreen.tsx

```diff
@@ -72,44 +77,46 @@ function AddMedicineScreen({navigation, route}: Props) {
             }}>
             <KeyboardAwareFlatList
               keyboardShouldPersistTaps={'handled'}
-              data={MEDICINES}
+              data={medicinesFiltered}
+              keyExtractor={(item) => item.name}
               renderItem={({item}) => {
-                if (
-                  !item.toLowerCase().includes(input.toLowerCase()) ||
-                  input === ''
-                ) {
-                  return null
-                }
-
```

Comment:
Is there an advantage is filtering the list before mapping over it?

Reply:
Yes definitely - we should never be returning null in the render item method of a flatlist. That's one of the reasons the list was showing a yellow box with warnings before - ie lots of nulls getting returned rather than react nodes with different keys.